src/js/angular/core/services/jwt-auth.service.js

```diff
@@ -3,294 +3,302 @@ import 'angular/core/services/openid-auth.service.js';
-             *                          In some cases this will be autodetected.
              * @param {boolean} noaccess If true a toast message about no access/config error will be shown.
-             *                           This overrides 'expired'.
              * @return {Promise<unknown>}
              */
-            $rootScope.redirectToLogin = function (expired, noaccess) {
-                if (jwtAuth.authTokenIsType('Bearer')) {
+            $rootScope.redirectToLogin = (expired, noaccess) => {
+                if (_authTokenIsType(TOKEN_TYPE_BEARER)) {
                     // OpenID login may be detected as expired either on initial validation
                     // when we get expired = true or indirectly via 401, setting expired = true
                     // handles the second case.
                     expired = true;
                 }
-                jwtAuth.clearAuthenticationInternal();
+                _clearAuthenticationInternal();
                 // remember where we were so we can return there
-                jwtAuth.updateReturnUrl();
+                updateReturnUrl();
 
-                $location.path('/login');
+                $location.path(LOGIN_PATH);
                 if (noaccess) {
-                    $location.search('noaccess');
+                    $location.search(NO_ACCESS);
                 } else if (expired) {
-                    $location.search('expired');
+                    $location.search(EXPIRED);
                 }
                 // Countering race condition. When the unauthorized interceptor catches error 401 or 409, then we must make
                 // sure that a request is made to access the login page before proceeding with the rejection of the
                 // original request. Otherwise the login page is not accessible in the context of spring security.
-                return new Promise(resolve => {
+                return new Promise((resolve) => {
                     setTimeout(() => {
                         resolve(true);
                     }, 100);
                 });
             };
-            $rootScope.hasExternalAuthUser = function () {
-                return jwtAuth.hasExternalAuthUser();
+
+            /**
+             * Returns whether the current user has external authentication.
+             * @return {boolean} True if the user is authenticated via an external system.
+             */
+            $rootScope.hasExternalAuthUser = () => hasExternalAuthUser();
+
+            // =========================
+            // Public functions
+            // =========================
+            /**
+             * Checks if the current user is authenticated.
+             * @return {boolean} True if the user is authenticated.
+             */
+            const isAuthenticated = () => !_securityEnabled || _hasExplicitAuthentication();
+
+            const _hasExplicitAuthentication = () => {
```

Comment:
What it the matter with these underscores BTW. In the context of classes and constructor functions it is a convention to denote private functions usually but here you've put them in the public section just because you then reexposed them via some other unprefixed functions. This clearly can be simplified.

Reply:
the goal was in the initial MR the changes to be minimal - to separate the private properties from the public ones. In the next iteration to be simplified. But this is already the subject of the next MR.